<?xml version="1.0"?>
<robot name="marker_cam" xmlns:xacro="http://ros.org/wiki/xacro">

    <xacro:include filename="$(find realsense2_description)/urdf/_d415.urdf.xacro" />
 
    <link name="world"/>
    <gazebo reference="world"/>

    <joint name="marker_cam_joint" type="fixed">
        <origin rpy="0 0 $(arg theta)" xyz="0 0 0"/>
        <parent link="world"/>
        <child link="marker_cam_link"/>
    </joint>

    <link name="marker_cam_link">
        <visual>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <mesh filename="package://nicol_rh8d/inventor/cam_holder.stl" scale="0.001 0.001 0.001"/>
        </geometry>
            <material name="cam_link_color">
                <color rgba="0.5 0.5 0.5 1.0"/>
            </material>
        </visual>
        <collision>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <geometry>
            <box size="0 0 0"/>
        </geometry>
        </collision>
        <inertial>
        <origin rpy="0 0 0" xyz="0 0 0"/>
        <mass value="0"/>
        <inertia ixx="0" ixy="0" ixz="0" iyy="0" iyz="0" izz="0"/>
        </inertial>
    </link>
    <gazebo reference="marker_cam_link">
        <material>Gazebo/Grey</material>
    </gazebo>

    <xacro:sensor_d415 parent="marker_cam_link" 
                  name="marker_realsense" 
                  use_nominal_extrinsics="true" 
                  use_mesh="true" >
        <origin xyz="$(arg x_dist) $(arg y_dist) $(arg z_dist)" rpy="0 ${pi*0.5} 0"/>
    </xacro:sensor_d415>
    <gazebo reference="marker_realsense">
        <material>Gazebo/Grey</material>
    </gazebo>

</robot>